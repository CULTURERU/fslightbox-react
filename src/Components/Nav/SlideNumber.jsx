import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { ifOnUpdateExistsHandleItForState } from "../../Helpers/State/ifOnUpdateExistsHandleItForState";

const SlideNumber = (
    {
        fsLightbox: {
            data: { totalSlides },
            componentsStates: { slide: slideState },
        }
    }
) => {
    const [slide, setSlide] = useState(1);
    slideState.get = () => slide;
    slideState.set = setSlide;

    useEffect(() => {
        ifOnUpdateExistsHandleItForState(slideState);
    });

    if (totalSlides === 1) {
        return null;
    }

    return (
        <div className="fslightbox-slide-number-container fslightbox-flex-centered">
            <div>{ slide }</div>
            <div className="fslightbox-slash">/</div>
            <div>{ totalSlides }</div>
        </div>
    );
};

SlideNumber.propTypes = {
    fsLightbox: PropTypes.object.isRequired
};
export default SlideNumber;