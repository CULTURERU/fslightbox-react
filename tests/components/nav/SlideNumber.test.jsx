import React from 'react';
import { shallow } from "enzyme";
import SlideNumber from "../../../src/components/nav/SlideNumber";
import { testComponentStateForStateChainAndFsLightbox } from "../../__tests-helpers__/testComponentStateForStateChainAndFsLightbox";

const fsLightbox = {
    componentsStates: {
        slideNumberUpdater: {}
    },
    data: {
        sourcesCount: 4
    },
    stageIndexes: {
        current: 0
    }

};
let slideNumber = shallow(<SlideNumber fsLightbox={ fsLightbox }/>);

describe('SlideNumber dom', () => {
    it('should match snapshot', () => {
        expect(slideNumber).toMatchSnapshot();
    });
});

describe('slideNumberUpdater component state', () => {
    testComponentStateForStateChainAndFsLightbox('slideNumberUpdater', fsLightbox);
});
