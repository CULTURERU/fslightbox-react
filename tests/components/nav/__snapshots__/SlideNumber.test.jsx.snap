// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideNumber DOM sourcesCount > 1 static DOM should match snapshot 1`] = `
<div
  className="fslightbox-slide-number-lightboxContainer fslightbox-flex-centered"
>
  <div>
    7
  </div>
  <div
    className="fslightbox-slash"
  >
    /
  </div>
  <div>
    5
  </div>
</div>
`;
